File: 347-top-k-frequent-elements/top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int,int>mp;
        vector<int>ans;
        for(int i:nums)
            ++mp[i];
        priority_queue<pair<int,int>>pq;
        int limit=mp.size()-k;
        for(auto it:mp)
            pq.push({it.second,it.first});
        while(ans.size()<k && !pq.empty())
        {
            ans.push_back(pq.top().second);
            pq.pop();
        }
        return ans;
    }
};

File: 347-top-k-frequent-elements/top-k-frequent-elements.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer,Integer>freqMap=new HashMap<>();
        for(int i:nums)
            freqMap.put(i,freqMap.getOrDefault(i,0)+1);
        PriorityQueue<Integer>pq=new PriorityQueue<>(
            (a,b)->{
                return freqMap.get(b)-freqMap.get(a);
        });
        for(int i:freqMap.keySet())
            pq.add(i);
        int ind=0;
        int ans[]=new int[k];
        while(ind<k && !pq.isEmpty())
            ans[ind++]=pq.poll();
        return ans;
    }
}